hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/container/SCMContainerManager.java

```diff
@@ -117,9 +119,28 @@ private void loadExistingContainers() throws IOException {
           ContainerInfoProto.PARSER.parseFrom(entry.getValue()));
       Preconditions.checkNotNull(container);
       containerStateManager.loadContainer(container);
-      if (container.getState() == LifeCycleState.OPEN) {
-        pipelineManager.addContainerToPipeline(container.getPipelineID(),
-            ContainerID.valueof(container.getContainerID()));
+      try {
+        if (container.getState() == LifeCycleState.OPEN) {
+          pipelineManager.addContainerToPipeline(container.getPipelineID(),
+              ContainerID.valueof(container.getContainerID()));
+        }
+      } catch (PipelineNotFoundException ex) {
+        LOG.warn("Found a Container {} which is in {} state with out a " +
+            "pipeline {}. Close Container.", container, container.getState(),
+            container.getPipelineID());
+
+        // Not firing CLOSE_CONTAINER event because CloseContainer Event
```

Comment:
This comment does not match with the code any more. Should we remove it? 
"// Not firing CLOSE_CONTAINER event ..."

Reply:
This was added to explain the reason why we are not calling CLOSE_CONTAINER event to close container. In future, if someone changes this part of the code, they will know the reason in doing this way.